rpmlint/checks/BinariesCheck.py

```diff
@@ -452,6 +454,16 @@ def _check_executable_shlib(self, pkg, pkgfile_path, path):
             if interp:
                 self.output.add_info('E', pkg, 'shared-library-not-executable', path)
 
+    def _check_optflags(self, pkg, pkgfile_path, path):
+        if not self.readelf_parser.is_archive:
+            mandatory_optflags = self.config.configuration['MandatoryOptflags']
+            if mandatory_optflags:
+                for dwarf_unit in self.objdump_parser.compile_units:
+                    tokens = dwarf_unit['producer'].split(' ')
+                    missing = [mo for mo in mandatory_optflags if mo not in tokens]
+                    if missing:
+                        self.output.add_info('W', pkg, 'missing-mandatory-optflags', path, ' '.join(missing))
```

Comment:
Should even be error I dare to say if it does not do too many false positives.

Reply:
I would wait with making it an error, we can do it later.